packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1624,14 +1624,13 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       value: string,
       next: boolean,
     ): ?N.Declaration {
+      // no declaration apart from enum can be followed by a line break.
```

Comment:
Not related to this PR: I think `enum` allowing line break after is a TS parser bug. We should report it to TypeScript teams.

Reply:
actual reason for this is how js threats those values

javascript has set of reserved keyword https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#keywords

`enum` is a `Future reserved keywords` and you can't use it as identifier in valid js code

Reply:
Ah makes sense. Concisely `interface` is also a _future reserved word_, but only in strict mode, while `enum` is always reserved.

Reply:
Should we parse `interface` like we do for `enum` when strict mode is enabled?

Reply:
No, we should always match the TS behavior: https://www.typescriptlang.org/play?alwaysStrict=false#code/DYUwLgBAlgdmICcBmBDAxiCBeCAmA3AFCGzzLoiECCEhA3gL5A